From 66c31956c909adb843a5c02558781163e1fc96cb Mon Sep 17 00:00:00 2001
From: "Wesley.CW Wang" <wesleycwwang@google.com>
Date: Mon, 22 Oct 2018 18:48:02 +0800
Subject: [PATCH 2/6] Fix EmergencyDialer UI when display size change

- Emergency shortcut button will use other size when display size is
larger/largest.
- Add values-sw345dp and values-sw360dp for
emergency shortcut button height/margin_vertical/text size.

Test: Manually
Bug: 117251739
Change-Id: I154ad8d798e8c2c0417bf1fcd33713d0c560ed47
---
 .../emergency_shortcut_buttons_group.xml      |  6 +++--
 res/values-sw345dp/dimens.xml                 | 24 +++++++++++++++++++
 res/values-sw360dp/dimens.xml                 | 24 +++++++++++++++++++
 res/values/dimens.xml                         | 11 ++++++---
 4 files changed, 60 insertions(+), 5 deletions(-)
 create mode 100644 res/values-sw345dp/dimens.xml
 create mode 100644 res/values-sw360dp/dimens.xml

diff --git a/res/layout/emergency_shortcut_buttons_group.xml b/res/layout/emergency_shortcut_buttons_group.xml
index bafedd1ec..7911f3070 100644
--- a/res/layout/emergency_shortcut_buttons_group.xml
+++ b/res/layout/emergency_shortcut_buttons_group.xml
@@ -38,7 +38,8 @@
                 android:ellipsize="end"
                 android:lineHeight="@dimen/emergency_number_title_line_height"
                 android:fontFamily="sans-serif-medium"
-                android:textAppearance="@style/SubtitleTextAppearance"
+                android:textColor="@android:color/white"
+                android:textSize="@dimen/emergency_number_title_text_size"
                 android:text="@string/single_emergency_number_title"/>
         </FrameLayout>
         <LinearLayout
@@ -63,7 +64,8 @@
                 android:maxLines="2"
                 android:ellipsize="end"
                 android:lineHeight="@dimen/location_text_line_height"
-                android:textAppearance="@style/SubtitleTextAppearance"/>
+                android:textColor="@android:color/white"
+                android:textSize="@dimen/emergency_location_text_size"/>
         </LinearLayout>
     </FrameLayout>
     <LinearLayout
diff --git a/res/values-sw345dp/dimens.xml b/res/values-sw345dp/dimens.xml
new file mode 100644
index 000000000..84db80ba1
--- /dev/null
+++ b/res/values-sw345dp/dimens.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2018 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License
+  -->
+
+<resources>
+    <dimen name="emergency_info_button_height">88dp</dimen>
+    <dimen name="emergency_info_button_margin_vertical">48dp</dimen>
+    <dimen name="emergency_shortcut_button_height">88dp</dimen>
+    <dimen name="emergency_number_title_text_size">12sp</dimen>
+    <dimen name="emergency_location_text_size">12sp</dimen>
+</resources>
\ No newline at end of file
diff --git a/res/values-sw360dp/dimens.xml b/res/values-sw360dp/dimens.xml
new file mode 100644
index 000000000..c57806522
--- /dev/null
+++ b/res/values-sw360dp/dimens.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2018 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License
+  -->
+
+<resources>
+    <dimen name="emergency_info_button_height">96dp</dimen>
+    <dimen name="emergency_info_button_margin_vertical">56dp</dimen>
+    <dimen name="emergency_shortcut_button_height">96dp</dimen>
+    <dimen name="emergency_number_title_text_size">14sp</dimen>
+    <dimen name="emergency_location_text_size">14sp</dimen>
+</resources>
\ No newline at end of file
diff --git a/res/values/dimens.xml b/res/values/dimens.xml
index b657e64c1..38241dea2 100644
--- a/res/values/dimens.xml
+++ b/res/values/dimens.xml
@@ -128,8 +128,8 @@
     <dimen name="emergency_number_title_group_padding_horizontal">16dp</dimen>
 
     <!-- Height and vertical margin for the emergency information button. -->
-    <dimen name="emergency_info_button_height">96dp</dimen>
-    <dimen name="emergency_info_button_margin_vertical">56dp</dimen>
+    <dimen name="emergency_info_button_height">80dp</dimen>
+    <dimen name="emergency_info_button_margin_vertical">40dp</dimen>
     <dimen name="emergency_info_button_fix_margin_vertical">40dp</dimen>
 
     <!-- Margin for the emergency information button text. -->
@@ -155,7 +155,7 @@
     <dimen name="location_image_width">16dp</dimen>
 
     <!-- The height for button of emergency shortcut. -->
-    <dimen name="emergency_shortcut_button_height">96dp</dimen>
+    <dimen name="emergency_shortcut_button_height">80dp</dimen>
 
     <!-- The height and width for the circle image of phone number type.-->
     <dimen name="phone_number_type_circle_image_height">40dp</dimen>
@@ -197,4 +197,9 @@
     <!-- The width for emergency number title container.-->
     <dimen name="emergency_number_title_container_width">210dp</dimen>
 
+    <!-- The text size for emergency number title.-->
+    <dimen name="emergency_number_title_text_size">13sp</dimen>
+
+    <!-- The text size for emergency location.-->
+    <dimen name="emergency_location_text_size">13sp</dimen>
 </resources>
-- 
2.17.1

